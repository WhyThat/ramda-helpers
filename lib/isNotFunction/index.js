import { complement } from 'ramda';
import isFunction from '../isFunction';

/**
 * Returns true if prop is not a function
 * @name isNotFunction
 * @see isFunction
 * @param {*} value
 * @returns {Boolean}
 * @example
 * isNotFunction(() => {}) //=> false
 * isNotFunction({}) //=> true
 */
const isNotFunction = complement(isFunction);
export default isNotFunction;
