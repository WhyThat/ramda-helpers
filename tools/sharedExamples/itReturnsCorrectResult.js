export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
