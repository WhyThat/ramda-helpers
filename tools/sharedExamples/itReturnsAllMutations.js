import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
