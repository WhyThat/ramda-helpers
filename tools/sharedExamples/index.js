
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
