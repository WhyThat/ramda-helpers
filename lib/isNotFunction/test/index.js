import { itReturnsAllMutations } from '../../../tools/sharedExamples';
import isNotFunction from '../';

describe('isNotFunction', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With array',
    props: [],
  }, {
    name: 'With object',
    props: {},
  }, {
    name: 'With function',
    props: () => {},
  }];

  itReturnsAllMutations(isNotFunction, mutations);
});

