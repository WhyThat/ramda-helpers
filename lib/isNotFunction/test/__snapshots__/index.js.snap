exports[`isNotFunction With array returns correct result 1`] = `true`;

exports[`isNotFunction With function returns correct result 1`] = `false`;

exports[`isNotFunction With null returns correct result 1`] = `true`;

exports[`isNotFunction With object returns correct result 1`] = `true`;
